src/main/java/com/atlassian/jira/cloud/jenkins/common/client/JenkinsAppApi.java

```diff
@@ -31,35 +31,24 @@
     private final ObjectMapper objectMapper;
     private static final int JWT_EXPIRY_SECONDS = 5 * 60;
 
-    private final PipelineLogger pipelineLogger;
-
-    public JenkinsAppApi(
-            final OkHttpClient httpClient,
-            final ObjectMapper objectMapper,
-            final PipelineLogger pipelineLogger) {
-        this.httpClient = Objects.requireNonNull(httpClient);
-        this.objectMapper = Objects.requireNonNull(objectMapper);
-        this.pipelineLogger = pipelineLogger;
-    }
-
     @Inject
     public JenkinsAppApi(final OkHttpClient httpClient, final ObjectMapper objectMapper) {
         this.httpClient = Objects.requireNonNull(httpClient);
         this.objectMapper = Objects.requireNonNull(objectMapper);
-        this.pipelineLogger = PipelineLogger.noopInstance();
```

Comment:
Moved the `pipelineLogger` from a field initialized in the constructor to a method parameter below to avoid attaching a pipelineLogger to an object that might be shared (because a PipelineLogger can't be shared between two different pipelines).